test/e2e/usersignup_test.go

```diff
@@ -45,662 +43,190 @@ func (s *userSignupIntegrationTest) TearDownTest() {
 
-		// Confirm the MUR was created
-		s.assertCreatedMUR(userSignup)
+		// then
+		s.checkUserSignupManualApproval()
 	})
 
-	s.T().Run("usersignup created with approved set to true", func(t *testing.T) {
-		// Create user signup - approval set to true
-		s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
-		userSignup, err := newUserSignup(s.awaitility.Host(), uuid.NewV4().String(), "robertjones@magenta.com")
-		require.NoError(s.T(), err)
+	s.T().Run("auto approval policy", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
 
-		userSignup.Spec.Approved = true
-		err = s.awaitility.Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(s.testCtx))
-		require.NoError(s.T(), err)
-		s.T().Logf("user signup '%s' created", userSignup.Name)
-
-		// Confirm that:
-		// 1) Confirm the UserSignup was created
-		// 2) the Approved condition is set to true
-		// 3) the Approved reason is set to ApprovedByAdmin
-		// 4) the Complete condition is set to true
-		userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(
-			v1alpha1.Condition{
-				Type:   v1alpha1.UserSignupApproved,
-				Status: corev1.ConditionTrue,
-				Reason: "ApprovedByAdmin",
-			},
-			v1alpha1.Condition{
-				Type:   v1alpha1.UserSignupComplete,
-				Status: corev1.ConditionTrue,
-			}))
-		require.NoError(s.T(), err)
+		s.T().Run("approved set to false", func(t *testing.T) {
+			// UserSignup created with approved flag set to false but it should be approved anyway
+			s.createUserSignupAndAssertAutoApproval(false)
+		})
 
-		// Confirm the MUR was created
-		s.assertCreatedMUR(userSignup)
+		s.T().Run("approved set to true", func(t *testing.T) {
+			// UserSignup created with approved flag set to true and it should be approved
+			s.createUserSignupAndAssertManualApproval(true)
+		})
 	})
 }
 
-func (s *userSignupIntegrationTest) assertCreatedMUR(userSignup *v1alpha1.UserSignup) {
-	mur, err := s.hostAwait.WaitForMasterUserRecord(userSignup.Status.CompliantUsername)
-	require.NoError(s.T(), err)
-
-	require.Len(s.T(), mur.Spec.UserAccounts, 1)
-	require.Equal(s.T(), userSignup.Name, mur.Labels["toolchain.dev.openshift.com/user-id"])
-	require.Equal(s.T(), userSignup.Name, mur.Spec.UserAccounts[0].Spec.UserID)
-	require.Equal(s.T(), "default", mur.Spec.UserAccounts[0].Spec.NSLimit)
-	require.NotNil(s.T(), mur.Spec.UserAccounts[0].Spec.NSTemplateSet)
-}
-
 func (s *userSignupIntegrationTest) TestTargetClusterSelectedAutomatically() {
-	// Set the user approval policy to automatic
+	// Create user signup
 	s.setApprovalPolicyConfig("automatic")
-
-	// Create user signup - no approval set
-	s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
-
-	userSignup, err := newUserSignup(s.awaitility.Host(), uuid.NewV4().String(), "reginald@alpha.com")
-	require.NoError(s.T(), err)
+	userSignup := newUserSignup(s.T(), s.awaitility.Host(), uuid.NewV4().String(), "reginald@alpha.com")
```

Comment:
I see that in all cases when  `newUserSignup` function is called it generates the UUID every time in the same way `uuid.NewV4().String()` - could we move it directly into the function `newUserSignup`?

Reply:
Yeah... we could for sure.
Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/47/commits/c99f78bfe01012cd5b9a996765a5f81c8e883b1e